### PR TITLE
add proxy warning for jenkins/git

### PR DESCRIPTION
**Pull request by gabemontero**
@bparees @bmcelvee 

to assist users regarding https://bugzilla.redhat.com/show_bug.cgi?id=1405952

### CODE REVIEW

File: dev_guide/builds/build_inputs.adoc [hunk 1]
```diff
@@ -257,6 +257,14 @@ xref:../../install_config/http_proxies.adoc#configuring-hosts-for-proxies-using-
 a global proxy for Git cloning using Ansible].
 endif::[]
 
+[NOTE]
+====
+For Pipeline strategy builds, given the current restrictions with the Git plug-in for Jenkins, proxies
+defined in your `BuildConfig` cannot be honored.  You must configure the proxy in the Jenkins UI.
+Instruction on how to configure proxies via the Jenkins UI can be found
+link:https://wiki.jenkins-ci.org/display/JENKINS/JenkinsBehindProxy[`here.`]
```
Comment (bparees) [inline dev_guide/builds/build_inputs.adoc]: it's really the git proxy (from the buildconfig) that's not honored (for purposes of accessing your git repo containing your Dockerfile).

env var proxies defined may (or may not) still be respected by various steps of your pipeline, depending on what those steps do and the plugins that implement them.  I think we need to be more specific here.

Comment (gabemontero) [inline dev_guide/builds/build_inputs.adoc]: attempt at clarification/cleanup pushed

File: dev_guide/builds/build_inputs.adoc [hunk 2]
```diff
@@ -257,6 +257,20 @@ xref:../../install_config/http_proxies.adoc#configuring-hosts-for-proxies-using-
...
+For Pipeline strategy builds, given the current restrictions with the Git plug-in for Jenkins, any Git
+operations via the Git plug-in will not leverage the HTTP or HTTPS proxy defined in the `BuildConfig`.
+The Git plug-in only will use the the proxy configured in the Jenkins UI at the Plugin Manager panel.
+This proxy will then be used for all git interactions within Jenkins, across all jobs.
+Instruction on how to configure proxies via the Jenkins UI can be found
+link:https://wiki.jenkins-ci.org/display/JENKINS/JenkinsBehindProxy[`here.`]
+
+The leveraging of either the proxy defined in the Jenkins UI, or the well know proxy
+enviornment variables (http_proxy, https_proxy, HTTP_PROXY, HTTPS_PROXY), varies step to
+step and plug-in to plug-in.
```
Comment (bparees) [inline dev_guide/builds/build_inputs.adoc]: i'm ok w/ dropping this second paragraph.  first one looks good.

Comment (gabemontero) [inline dev_guide/builds/build_inputs.adoc]: cool - 2nd paragraph dropped

File: dev_guide/builds/build_inputs.adoc [hunk 3]
```diff
@@ -257,6 +257,16 @@ xref:../../install_config/http_proxies.adoc#configuring-hosts-for-proxies-using-
 a global proxy for Git cloning using Ansible].
 endif::[]
 
+[NOTE]
+====
+For Pipeline strategy builds, given the current restrictions with the Git plug-in for Jenkins, any Git
+operations via the Git plug-in will not leverage the HTTP or HTTPS proxy defined in the `BuildConfig`.
+The Git plug-in only will use the the proxy configured in the Jenkins UI at the Plugin Manager panel.
+This proxy will then be used for all git interactions within Jenkins, across all jobs.
+Instruction on how to configure proxies via the Jenkins UI can be found
```
Comment (bparees) [inline dev_guide/builds/build_inputs.adoc]: s/Instruction/Instructions/

(but maybe that's a personal quirk?  i'm not sure Instruction is actually wrong.  we can let @bmcelvee decide)

**Comment by bparees** [comment]
lgtm pending instruction(s) resolution.

File: dev_guide/builds/build_inputs.adoc (hunk 3 above)
Comment (bmcelvee) [inline dev_guide/builds/build_inputs.adoc]: @bparees @gabemontero I agree with "Instructions". Gabe, I'll go ahead and merge/open a quick follow-up so you don't have to worry about changing anything.

**Comment by bmcelvee** [comment]
[rev_history]
|xref:../dev_guide/builds/build_inputs.adoc#dev-guide-build-inputs[Build Inputs]
|Added proxy warning for Jenkins in the xref:../dev_guide/builds/build_inputs.adoc#using-a-proxy-for-git-cloning[Using a Proxy] section.
%

File: dev_guide/builds/build_inputs.adoc (hunk 3 above)
Comment (gabemontero) [inline dev_guide/builds/build_inputs.adoc]: cool thanks @bmcelvee !

**Pull request merged**